src/options-processing.cc

```diff
@@ -75,6 +75,7 @@ void service_dir_opt::build_paths(bool am_system_init)
         }
 
         done_user_home:
+        service_dir_paths.emplace_back("/etc/dinit.d/user", /*dyn_allocd=*/false);
```

Comment:
Adding the directory here unconditionally then has a problem in case the checks above failed (eg `getenv("XDG_CONFIG_HOME")` and `get_user_home()` both returned `nullptr`). The current behaviour - which perhaps isn't ideal - is to fall back on using the system manager paths. By adding the new user path here without any sort of guard, in that case _both_ `/etc/dinit.d/user` and the system paths (`/etc/dinit.d` etc as per below) will be added to the service directory path list, which I think is undesirable.

At least adding the user dirs should be guarded with a check for `home_service_dir_set`.